editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -149,26 +151,140 @@ void EditorQuickOpenDialog::popup_dialog(const Vector<StringName> &p_base_types,
 	ERR_FAIL_COND(p_base_types.is_empty());
 	ERR_FAIL_COND(!p_item_selected_callback.is_valid());
 
+	property_object = nullptr;
+	property_path = "";
 	item_selected_callback = p_item_selected_callback;
 
 	container->init(p_base_types);
+	container->set_instant_preview_toggle_visible(false);
 	get_ok_button()->set_disabled(container->has_nothing_selected());
 
 	set_title(get_dialog_title(p_base_types));
 	popup_centered_clamped(Size2(780, 650) * EDSCALE, 0.8f);
 	search_box->grab_focus();
 }
 
-void EditorQuickOpenDialog::ok_pressed() {
-	item_selected_callback.call(container->get_selected());
+void EditorQuickOpenDialog::popup_dialog_for_property(const Vector<StringName> &p_base_types, Object *p_obj, String p_path) {
+	ERR_FAIL_COND(p_base_types.is_empty());
+	property_object = p_obj;
+	property_path = p_path;
+	initial_property_value = property_object->get(property_path);
 
-	container->save_selected_item();
+	container->init(p_base_types);
+	container->set_instant_preview_toggle_visible(true);
+	get_ok_button()->set_disabled(container->has_nothing_selected());
+
+	set_title(get_dialog_title(p_base_types));
+	popup_centered_clamped(Size2(780, 650) * EDSCALE, 0.8f);
+	search_box->grab_focus();
+}
+
+void EditorQuickOpenDialog::ok_pressed() {
+	update_property();
 	container->cleanup();
 	search_box->clear();
 	hide();
 }
 
+void EditorQuickOpenDialog::selection_changed() {
+	if (!EDITOR_GET("filesystem/quick_open_dialog/instant_preview") || property_object == nullptr) {
```

Comment:
You should check whether the button is pressed (and visible), instead of checking the setting every time.